public/app/plugins/datasource/elasticsearch/elastic_response.ts

```diff
@@ -522,3 +461,110 @@ export class ElasticResponse {
     return { data: dataFrame };
   }
 }
+
+type Doc = {
+  _id: string;
+  _type: string;
+  _index: string;
+  _source?: any;
+};
+
+/**
+ * Flatten the docs from response mainly the _source part which can be nested. This flattens it so that it is one level
+ * deep and the keys are: `level1Name.level2Name...`. Also returns list of all properties from all the docs (not all
+ * docs have to have the same keys).
+ * @param hits
+ */
+const flattenHits = (hits: Doc[]): { docs: Record<string, any>[]; propNames: string[] } => {
```

Comment:
This and the later functions were just moved out of the main function for a bit more clarity.